lib/rspec/support/reentrant_mutex.rb

```diff
@@ -28,8 +28,8 @@ def synchronize
     private
 
       def enter
-        @mutex.lock if @owner != Thread.current
-        @owner = Thread.current
```

Comment:
Do we need to `require 'fiber'` somewhere? `Fiber.current` errors out in `irb -f`.

Reply:
Oh yeah probably.